test/strscan/test_stringscanner.rb

```diff
@@ -206,6 +206,19 @@ def test_pos_unicode
     assert_equal 11, s.charpos
   end
 
+  def test_charpos_avoid_segfault
+    string = +'ruby'
+    scanner = create_string_scanner(string)
+    pre = Module.new do
+      def byteslice(*args)
+      end
+    end
+    string.singleton_class.prepend(pre)
+    assert_nothing_raised do
+      scanner.charpos
+    end
```

Comment:
Please don't use `assert_nothing_raised` as much as possible. Because it doesn't provide new check. If we don't have `assert_nothing_raised`, exception is processed by testing framework.

We should use normal assertions such as `assert_equal` instead of using `assert_nothing_raised`.

How about the following?

```suggestion
    assert_equal(0, scanner.charpos)
```

Reply:
Oops! You perfectly right... 🙇  I basically avoid `assert_nothing_raised` and rspec's `not_to raise_error` ways, I dislike them 😅 
I guess I was really panicked in this PR...

Applied in https://github.com/ruby/strscan/pull/20/commits/89e7b337f91baa1714241b01a37e50b0865551e1 🙏